cmd/mirror-main.go

```diff
@@ -129,6 +129,10 @@ var (
 			Name:  "monitoring-address",
 			Usage: "if specified, a new prometheus endpoint will be created to report mirroring activity. (eg: localhost:8081)",
 		},
+		cli.BoolFlag{
```

Comment:
Q: Would an integer make sense? 0=no retry; n = n retries; -n = infinite retries?

Right now you just decided that 3 is the magic number :)

Reply:
@kannappanr askes me not to make the flag user configurable, but rather set a number somewhere below 5. Making the number configurable is no problem.